src/Extensions/Microsoft.Diagnostics.Monitoring.Extension.Common/EgressHelper.cs

```diff
@@ -35,8 +35,15 @@ private static async Task<int> Egress<TOptions>(EgressProvider<TOptions> provide
             EgressArtifactResult result = new();
             try
             {
-                string jsonConfig = Console.ReadLine();
-                ExtensionEgressPayload configPayload = JsonSerializer.Deserialize<ExtensionEgressPayload>(jsonConfig);
+                StdInStream = Console.OpenStandardInput();
+
+                byte[] payloadLengthBuffer = new byte[sizeof(long)]; // Stream's Position is a long
+                _ = StdInStream.Read(payloadLengthBuffer);
```

Comment:
See https://github.com/dotnet/runtime/blob/f9ebd42d05ca88f086d472832487fbccfc7fe4ac/src/libraries/System.Private.CoreLib/src/System/IO/Stream.cs#L428 for how to write the read loop (doens't have to be exactly this, but gives you the idea on how to check whether more needs to be read).

For .NET 7 and above, you can use [Stream.ReadExactly](https://learn.microsoft.com/en-us/dotnet/api/system.io.stream.readexactly?view=net-8.0) instead of writing your own read loop.